live-hls.js

```diff
@@ -114,11 +128,40 @@ getHeaderObjects = function(fileContent) {
   return header;
 };
 
+getSegmentHeader = function(lines, index) {
+  var i;
+  var header='';
+  var byterange='';
+  var segment;
+  for(i=(index-1);i>0;i--) {
+    //Search backwards from index to get all header lines
+    if (lines[i].indexOf('EXTINF')>-1) {
+      header=lines[i];
+    } else if (lines[i].indexOf('EXT-X-BYTERANGE') > -1) {
+      byterange=lines[i];
+    } else
+    {
+      break;
+    }
+  }
+  segment = {
+    tsfile: lines[index]
+  };
+  if (byterange!='') {
+    segment.byterange=byterange;
+  }
+  if (header!='') {
+    segment.header=header;
+  }
+  return segment;
+};
+
 getResources = function(fileContent, request) {
   var lines = fileContent.split('\n');
   var header;
   var file;
   var resource = [];
```

Comment:
I'm actually not sure what our current style on declarations is.  I know having them all one var used to be the preferred style, but I have heard talk on a reversal of this policy.  I'll ask about it.